### PR TITLE
feat: template options refinements

### PR DESCRIPTION
**Pull request by guybedford**
This PR implements a number of renamings and adjustments to the template system, fixing a number of issues in the process.

* As a rule, all properties in the `chompfile.toml` are now `kebab-case` per convention.
* `[task.args]` for setting template options has been renamed to `[task.template-opts]` to make it clearer what this is doing.
* A new `[default-template-opts.template-name]` top-level setting is now provided for setting default options by template. Resolves https://github.com/guybedford/chomp/issues/27.
* `autoInstall` is now `auto-install`, and `PKG_MANAGER` is no longer an environment variable but rather a template option `pkg-manager`
* The `jspm:generate` template is renamed to just `jspm`.

This PR also adopts a manual chaining of template options within the template definitions themselves, although that is mostly an implementation detail.

This PR also includes various template fixes including fixing https://github.com/guybedford/chomp/issues/29 and some normalization in the JSPM generator and Babel plugins.

TLDR; here's a before after configuration example:

BEFORE:

```toml
version = 0.1

[env]
  PKG_MANAGER = "pnpm"

[[task]]
  template = "jspm"
  [task.args]
    autoInstall = true
    env = ["browser", "production"]
```

AFTER:

```toml
version = 0.1

[default-template-opts.npm]
  auto-install = true
  pkg-manager = 'pnpm'

[[task]]
  template = "jspm"
  [task.template-opts]
    env = ["browser", "production"]
```

While this is strictly a breaking change, since no release has yet been made there is no version bump.

### CODE REVIEW

**Comment by canadaduane** [comment]
What is the `.npm` suffix on `[default-template-opts.npm]`?

**Comment by guybedford** [comment]
This means these are the default options for the npm template.

**Comment by canadaduane** [comment]
But I'm not using the npm template? Or am I? And if I am, how do I know?

**Comment by canadaduane** [comment]
To be clear: I think it's confusing that it says `template = "jspm"` right below, but we are configuring `npm` template.

**Comment by guybedford** [comment]
Templates are functions that output a list of tasks, where those tasks in
turn can use templates. The jspm template will output the call to the
necessary npm template for JSPM to work, with the global option making auto
install configurable.

Would calling it global-options.npm seem clearer? It can be useful to eg
configure a global Babel config this way for all Babel templates to avoid
repetition so that’s why I was making sure that connection was clear first
and foremost.
